src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -221,24 +249,169 @@ public void onTraversalStart() {
      */
     @Override
     public void traverse() {
-        try {
-            importer.importBatch();
-        } catch (IOException e) {
-            throw new UserException("GenomicsDB importBatch raised exception", e);
+
+        if (batchSize == DEFAULT_ZERO_BATCH_SIZE) {
+            logger.info("GenomicsDB import batch started");
+            long variantContextBufferSize = vcfBufferSizePerSample * sampleNames.size();
+
+            Map<String, FeatureReader<VariantContext>> sampleToReaderMap = getFeatureReaders(sampleNames, sampleNameToFileName, 0, sampleNames.size());
+
+            GenomicsDBImportConfiguration.ImportConfiguration importConfiguration =
+                createImportConfiguration(workspace, arrayName,
+                    variantContextBufferSize, segmentSize,
+                    vidMapJSONFile.getAbsolutePath(), callsetMapJSONFile.getAbsolutePath(),
+                    0, sampleToReaderMap.size() - 1);
+
+            try {
+                importer = new GenomicsDBImporter(sampleToReaderMap, mergedHeader, intervals.get(0), importConfiguration);
+            } catch (IOException e) {
+                throw new UserException("Error initializing GenomicsDBImporter", e);
+            }
+
+            try {
+                importer.importBatch();
+                assert importer.isDone();
+            } catch (IOException e) {
+                throw new UserException("GenomicsDB importBatch raised exception", e);
+            }
+
+            closeReaders(sampleToReaderMap);
+
+        } else {
```

Comment:
Extract the body of the `else` here to a `doBatchedImport()` method, and call it from here.